includes/Classifai/Providers/OpenAI/Embeddings.php

```diff
@@ -0,0 +1,722 @@
+
+use Classifai\Providers\Provider;
+use Classifai\Providers\OpenAI\APIRequest;
+use Classifai\Providers\OpenAI\Tokenizer;
+use Classifai\Providers\OpenAI\EmbeddingCalculations;
+use Classifai\Providers\Watson\NLU;
+use Classifai\Watson\Normalizer;
+use function Classifai\get_asset_info;
+use function Classifai\language_processing_features_enabled;
+use WP_Error;
+
+class Embeddings extends Provider {
+
+	use \Classifai\Providers\OpenAI\OpenAI;
+
+	/**
+	 * OpenAI Embeddings URL
+	 *
+	 * @var string
+	 */
+	protected $api_url = 'https://api.openai.com/v1/embeddings';
+
+	/**
+	 * OpenAI Embeddings model
+	 *
+	 * @var string
+	 */
+	protected $model = 'text-embedding-ada-002';
+
+	/**
+	 * Maximum number of tokens our model supports
+	 *
+	 * @var int
+	 */
+	protected $max_tokens = 8191;
+
+	/**
+	 * OpenAI Embeddings constructor.
+	 *
+	 * @param string $service The service this class belongs to.
+	 */
+	public function __construct( $service ) {
+		parent::__construct(
+			'OpenAI Embeddings',
+			'Embeddings',
+			'openai_embeddings',
+			$service
+		);
+
+		// Set the onboarding options.
+		$this->onboarding_options = array(
+			'title'    => __( 'OpenAI Embeddings', 'classifai' ),
+			'fields'   => array( 'api-key' ),
+			'features' => array(
+				'enable_classification' => __( 'Classify content within existing term structure', 'classifai' ),
+			),
+		);
+	}
+
+	/**
+	 * Can the functionality be initialized?
+	 *
+	 * @return bool
+	 */
+	public function can_register() {
+		$settings = $this->get_settings();
+
+		if ( empty( $settings ) || ( isset( $settings['authenticated'] ) && false === $settings['authenticated'] ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Register what we need for the plugin.
+	 *
+	 * This only fires if can_register returns true.
+	 */
+	public function register() {
+		$settings = $this->get_settings();
+
+		if ( isset( $settings['enable_classification'] ) && 1 === (int) $settings['enable_classification'] ) {
+			add_action( 'wp_insert_post', [ $this, 'generate_embeddings_for_post' ] );
+			add_action( 'created_term', [ $this, 'generate_embeddings_for_term' ] );
+			add_action( 'edited_terms', [ $this, 'generate_embeddings_for_term' ] );
+			add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_editor_assets' ], 9 );
+			add_filter( 'rest_api_init', [ $this, 'add_process_content_meta_to_rest_api' ] );
+		}
+	}
+
+	/**
+	 * Enqueue editor assets.
+	 */
+	public function enqueue_editor_assets() {
+		global $post;
+
+		if ( empty( $post ) ) {
+			return;
+		}
+
+		wp_enqueue_script(
+			'classifai-gutenberg-plugin',
+			CLASSIFAI_PLUGIN_URL . 'dist/gutenberg-plugin.js',
+			get_asset_info( 'gutenberg-plugin', 'dependencies' ),
+			get_asset_info( 'gutenberg-plugin', 'version' ),
+			true
+		);
+
+		wp_add_inline_script(
+			'classifai-gutenberg-plugin',
+			sprintf(
+				'var classifaiEmbeddingData = %s;',
+				wp_json_encode(
+					[
+						'enabled'              => true,
```

Comment:
Maybe this should reflect the actual enabled value instead of the hard-coded true value.

Reply:
So we never make it to this point if the feature is not enabled, which is why I left this hardcoded instead of pulling the value from our settings again